From: Kemeng Shi <shikemeng@huaweicloud.com>
Subject: mm/page_ext: remove rollback for untouched mem_section in online_page_ext
Date: Fri, 14 Jul 2023 19:47:48 +0800

If init_section_page_ext failed, we only need rollback for mem_section
before failed mem_section.  Make rollback end point to failed mem_section
to remove unnecessary rollback.

As pfn += PAGES_PER_SECTION will be executed even if init_section_page_ext
failed.  So pfn points to mem_section after failed mem_section.  Subtract
one mem_section from pfn to get failed mem_section.

Link: https://lkml.kernel.org/r/20230714114749.1743032-3-shikemeng@huaweicloud.com
Signed-off-by: Kemeng Shi <shikemeng@huaweicloud.com>
Cc: Joonsoo Kim <iamjoonsoo.kim@lge.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_ext.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/page_ext.c~mm-page_ext-remove-rollback-for-untouched-mem_section-in-online_page_ext
+++ a/mm/page_ext.c
@@ -424,6 +424,7 @@ static int __meminit online_page_ext(uns
 		return 0;
 
 	/* rollback */
+	end = pfn - PAGES_PER_SECTION;
 	for (pfn = start; pfn < end; pfn += PAGES_PER_SECTION)
 		__free_page_ext(pfn);
 
_
